core/domain/exp_services_test.py

```diff
@@ -1237,6 +1237,94 @@ def test_apply_change_list(self) -> None:
             changed_exploration_objective.objective,
             'new objective')
 
+    def test_apply_change_list_handles_invalid_cmd(self) -> None:
+        self.save_new_linear_exp_with_state_names_and_interactions(
+            self.EXP_0_ID, self.owner_id, ['State 1', 'State 2'],
+            ['TextInput'], category='Algebra',
+            correctness_feedback_enabled=True)
+        class MockExplorationChange:
+            cmd: str = ''
+            property_name: str = ''
+            state_name: str = ''
+            new_value: str = ''
```

Comment:
Since we always set `self.cmd = cmd`, this doesn't appear to do anything.

Also, defining class variables like this can be dangerous because they're shared across all instances of the class. That can lead to unexpected behavior when they are mutable

ditto elsewhere

Reply:
Fixed